backend/src/backend/database/__init__.py

```diff
@@ -33,11 +35,18 @@ def my_loads(s, *args, **kwargs):
     )
 
 
-def get_session() -> OrmSession:
-    """The get_session() function is a utility designed to facilitate working with
+def gen_session() -> Generator[OrmSession, None, None]:
+    """The gen_session() function is a utility designed to facilitate working with
     an Object-Relational Mapping (ORM) system.
     It allows users to obtain a session object that is essential for performing various
     database operations using the ORM framework.
     """
     with Session.begin() as session:
-        yield session
+        try:
+            yield session
+            session.commit()
+        except Exception as e:
+            logger.info(e)
```

Comment:
Try not to introduce features inside a review behind a _Fix_ commit ; it's hard to understand the motive.

If you are to commit and close manuall, you don't need the contextmanager at all.

Also, use the appropriate log level. Info is not appropriate for such an error. You can use an `error()` and even an `exception()` passing it the exception so you get the stack trace